src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java

```diff
@@ -0,0 +1,127 @@
+package com.imagination.cbs.service.helper;
+
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import com.imagination.cbs.domain.ApprovalStatusDm;
+import com.imagination.cbs.domain.Booking;
+import com.imagination.cbs.domain.BookingRevision;
+import com.imagination.cbs.exception.CBSApplicationException;
+import com.imagination.cbs.exception.CBSUnAuthorizedException;
+import com.imagination.cbs.security.CBSUser;
+import com.imagination.cbs.service.AdobeSignService;
+import com.imagination.cbs.service.Html2PdfService;
+import com.imagination.cbs.service.LoggedInUserService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BookingHrApproveHelperTest {
+
+	@InjectMocks
+	private BookingHrApproveHelper bookingHrApproveHelper;
+	
+	@Mock
+	private LoggedInUserService loggedInUserService;
+
+	@Mock
+	private Html2PdfService html2PdfService;
+
+	@Mock
+	private AdobeSignService adobeSignService;
+
+	@Mock
+	private BookingSaveHelper bookingSaveHelper;
+
+	@Mock
+	private EmailHelper emailHelper;
+	
+	@Test
+	public void shouldPrepareMailAndSendToHrWithBookingRevisionDetailsWhenUserIsAuthorized() throws IOException {
+		Booking booking = createBooking();
+		BookingRevision bookingRevision = createBookingRevision();
+		byte b[] = {20,10,30,5};
+		ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+		byteArrayOutputStream.write(b);
+		CBSUser cbsUser = createCBSUser();
+		
+		when(loggedInUserService.isCurrentUserInHRRole()).thenReturn(true);
+		when(bookingSaveHelper.getLatestRevision(booking)).thenReturn(bookingRevision);
+		when(html2PdfService.generateAgreementPdf(bookingRevision)).thenReturn(byteArrayOutputStream);
+		when(adobeSignService.uploadAndCreateAgreement((Mockito.any(InputStream.class)),Mockito.eq("service.pdf"))).thenReturn("CBS-111");
+		when(adobeSignService.sendAgreement("CBS-111", bookingRevision)).thenReturn("C-546");
+		when(loggedInUserService.getLoggedInUserDetails()).thenReturn(cbsUser);
+		when(bookingSaveHelper.saveBooking(booking, bookingRevision, 1006L,cbsUser)).thenReturn(booking);
+		
+		bookingHrApproveHelper.hrApprove(booking);
+		
+		verify(loggedInUserService).isCurrentUserInHRRole();
```

Comment:
these all should be tested in separate test cases.

Reply:
This is the part of one condition, cant be tested in seperate test cases.

Reply:
what i mean here by separate test case is to put every verify in different test, i m not sure why its not possible ? if you feel trouble then ping me .. i will help on this.

Reply:
all your when code we moved to a common or before method, so it wont be duplicate in every test case. we are trying to test the code in units ... thats whey our test cases should be small and oriented.

Reply:
sure, tomorrow i will ping you and we discuss on this.